src/ShopBundle/objects/db/TShopOrder.class.php

```diff
@@ -540,6 +543,32 @@ protected function SaveOrderNotificationDataToOrder($bOrderSend, $oMail, $sSendT
         $this->AllowEditByAll(false);
     }
 
+    protected function SetSendOrderNotificationState(bool $orderSend): void
+    {
+        if (true === TGlobal::IsCMSMode()) {
+            return;
+        }
+
+        $aData = $this->sqlData;
+        if (true === $orderSend) {
+            $aData = $this->AddOrderNotificationDataForOrderSuccess($aData);
+        } else {
+            $aData = $this->AddOrderNotificationDataForOrderNotDone($aData);
+        }
+
+        $this->LoadFromRow($aData);
+        $this->AllowEditByAll(true);
+        $this->Save();
```

Comment:
This saves the data to the database to "inform" another request?

Could this other request be faster than this save?

Reply:
Yes the first request should set the flag. So the next request does not send the mail again.